server/src/main/java/org/apache/iotdb/db/utils/FilePathUtils.java

```diff
@@ -46,4 +54,82 @@ public static String regularizePath(String filePath){
     return resource.getFile().getAbsolutePath().split(PATH_SPLIT_STRING);
   }
 
+  /**
+   * get paths from group by level, like root.sg1.d2.s0, root.sg1.d1.s1
+   * level=1, return [root.sg1, 0] and pathIndex turns to be [[0, root.sg1], [1, root.sg1]]
+   * @param rawPaths
+   * @param level
+   * @param pathIndex
+   * @return
+   */
+  public static Map<String, Long> getPathByLevel(List<Path> rawPaths, int level, Map<Integer, String> pathIndex) {
+    // pathGroupByLevel -> count
+    Map<String, Long> finalPaths = new TreeMap<>();
+
+    int i = 0;
+    for (Path value : rawPaths) {
+      String[] tmpPath = value.getFullPath().split("\\.");
```

Comment:
Hey, just like @qiaojialin said, you can reuse MetaUtils.getNodesName() method here~